### ISSUE TITLE
[CLIENT_FETCH_ERROR] with next-auth and v12.2.5-canary.4

### ISSUE THREAD

**Issue by awareness481** [vercel/next.js#39495]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

    Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 20.6.0: Tue Feb 22 21:10:42 PST 2022; root:xnu-7195.141.26~1/RELEASE_ARM64_T8101
    Binaries:
      Node: 16.15.1
      npm: 8.11.0
      Yarn: N/A
      pnpm: 7.6.0
    Relevant packages:
      next: 12.2.5-canary.4
      eslint-config-next: N/A
      react: 18.2.0
      react-dom: 18.2.0

### Describe the Bug

When trying to login, an error appears. Not happening in earlier canary versions. (can only repro in v12.2.5-canary.4)

```
{
    "error": {
        "message": "Unexpected token '<', \"<!DOCTYPE \"... is not valid JSON",
        "stack": "SyntaxError: Unexpected token '<', \"<!DOCTYPE \"... is not valid JSON",
        "name": "SyntaxError"
    },
    "url": "/api/auth/session",
    "message": "Unexpected token '<', \"<!DOCTYPE \"... is not valid JSON",
    "client": true
}
```
ref: https://github.com/nextauthjs/next-auth/issues/5008#issuecomment-1211666440

### Expected Behavior

To be able to login

### Link to reproduction

https://github.com/awareness481/next-auth-v12.2.5-canary.4

### To Reproduce

1. Clone https://github.com/awareness481/next-auth-v12.2.5-canary.4 (next-auth example with canary version)
2. Install deps
3. Try to login
4. Observe error

(Edit: If you dont have at least one provider setup, you ll get a different error message but it is irrelevant)

**Comment by awareness481** [comment]
**Update**

I can confirm that this is the PR that is causing the issues https://github.com/vercel/next.js/pull/38483 (although weirdly enough, it seems to work for most envs)

To confirm this I:

1. Setup a local next.js, with the latest version (v12.2.5-canary.4)
2. Run `git revert -m 1 4e6d05575ee6c0c4f78130519f1d3f198bc05084` (reverts above PR)
3. Setup my repro repo to work with the local next.js
4. Setup google envs and provider & was able to log in

**Comment by ijjk** [comment]
Hi, thanks for the detailed info, this should be fixed in `v12.2.5-canary.5`, please update and give it a try!
